src/numerical_checks.jl

```diff
@@ -82,6 +82,98 @@ function check_inverse_retraction(
+
+@doc raw"""
+    check_geodesic(
+        M::AbstractManifold,
+        p=rand(M),
+        X=rand(M; vector_at=p);
+        #
+        tol::Real = 1e-12,
+        io::Union{IO,Nothing} = nothing,
+        N::Int = 101,
+        name::String = "geodesic",
+        plot::Bool = false,
+        error::Symbol = :none,
+        inverse_retraction_method::AbstractInverseRetractionMethod = default_inverse_retraction(M),
+        vector_transport_method::AbstractVectorTransportMethod = default_vector_transport(M),
+    )
+
+Numerically check whether the [`geodesic`](@ref) implementation is correct.
+This check requires both the [`log`](@ref) and [`parallel_transport_to`](@ref) functions to be implemented
+in order to an exact numerical verification.
+We further require the [`norm`](@ref) function to be implemented for the [`AbstractManifold`](@ref) `M`.
+
+You can provide an `inverse_retraction_method` and `vector_transport_method` to be used,
+but the verification will then be less accurate.
+
+The tests performed are the following based on sampling the geodesic ``\gamma(t) = ``[`geodesic`](@ref)`(M, p, X)` at
+`N` equidistant points ``t_i = i/(N-1)``, `i=0,...,N-1` we denote by ``p_i = \gamma(t_i)``
+
+1. Compute the norms of the tangent vectors ``\dot\gamma(t_i) ≈ X_i = \log_{p_i}(p_{i+1})``, ``i=0,...,N-2``
+   and check how far these norms deviate from their mean value.
+2. Check how far this mean deviates from being ``\frac{||X||}{N-1}``
+3. Check that the parallel transport of each ``X_i`` from ``p_i`` to ``p_{i+1}`` is close to ``X_{i+1}``.
+
+# Arguments
+* `M`:    the manifold to check
+* `p`:    point on the manifold to start the geodesic, a random point is used by default
+* `X`:    tangent vector at `p` to start the geodesic, a random tangent vector is used by default
+
+# Keyword arguments
+* `tol`:     if all errors are below this tolerance, the geodesic is considered to be exact
+* `io`:      provide an `IO` to print the result to
+* `N`:       number of points to sample the geodesic on ``[0,1]`` (default: 101)
+* `plot`:    whether to plot the result (if `Plots.jl` is loaded).
+* `error`:   specify how to report errors: `:none`, `:info`, `:warn`, or `:error` are available
+* `inverse_retraction_method`:  method to use for the inverse retraction, it is recommended to use [`LogarithmicInverseRetraction`](@ref)
+* `vector_transport_method`:    method to use for the vector transport, it is recommended to use [`ParallelTransport`](@ref)
+"""
+function check_geodesic(
+        M::AbstractManifold,
+        p = rand(M),
+        X = rand(M; vector_at = p);
+        tol::Real = 1.0e-12,
+        io::Union{IO, Nothing} = nothing,
+        N::Int = 101,
+        plot::Bool = false,
+        error::Symbol = :none,
+        inverse_retraction_method::AbstractInverseRetractionMethod = default_inverse_retraction_method(M),
+        vector_transport_method::AbstractVectorTransportMethod = default_vector_transport_method(M),
```

Comment:
Maybe it would be a bit nicer to have `LogarithmicInverseRetraction()` and `ParallelTransport()` as defaults? The tested properties aren't guaranteed otherwise.

Reply:
Sure, no problem; I was unsure about that default as well; now documented that more thoroughly and switched to log and PT